integrations/venice-duckdb/src/main/java/com/linkedin/venice/sql/DuckDBDaVinciRecordTransformer.java

```diff
@@ -0,0 +1,130 @@
+  private final Set<String> primaryKeys = new HashSet<>();
+
+  public DuckDBDaVinciRecordTransformer(int storeVersion, boolean storeRecordsInDaVinci, String baseDir) {
+    super(storeVersion, storeRecordsInDaVinci);
+    versionTableName = baseVersionTableName + storeVersion;
+    duckDBUrl = "jdbc:duckdb:" + baseDir + "/" + duckDBFilePath;
+    primaryKeys.add(primaryKey);
+  }
+
+  @Override
+  public Schema getKeySchema() {
+    return Schema.create(Schema.Type.STRING);
+  }
+
+  // ToDo: Put real schema
+  @Override
+  public Schema getOutputValueSchema() {
+    Schema valueSchema = SchemaBuilder.record("nameRecord")
+        .namespace("example.avro")
+        .fields()
+        .name("firstName")
+        .type()
+        .stringType()
+        .stringDefault("")
+        .name("lastName")
+        .type()
+        .stringType()
+        .stringDefault("")
+        .endRecord();
+    return valueSchema;
+  }
+
+  @Override
+  public DaVinciRecordTransformerResult<GenericRecord> transform(Lazy<String> key, Lazy<GenericRecord> value) {
+    // Record transformation happens inside processPut as we need access to the connection object to create the prepared
+    // statement
+    return new DaVinciRecordTransformerResult<>(DaVinciRecordTransformerResult.Result.UNCHANGED);
+  }
+
+  @Override
+  public void processPut(Lazy<String> key, Lazy<GenericRecord> value) {
+    Schema valueSchema = value.get().getSchema();
+    String upsertStatement = AvroToSQL.upsertStatement(versionTableName, valueSchema);
+
+    try (Connection connection = DriverManager.getConnection(duckDBUrl)) {
+      BiConsumer<GenericRecord, PreparedStatement> upsertProcessor = AvroToSQL.upsertProcessor(valueSchema);
+
+      try (PreparedStatement preparedStatement = connection.prepareStatement(upsertStatement)) {
+        upsertProcessor.accept(value.get(), preparedStatement);
+      }
+    } catch (SQLException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void processDelete(Lazy<String> key) {
+    try (Connection connection = DriverManager.getConnection(duckDBUrl);
+        Statement stmt = connection.createStatement()) {
+      String sqlDelete = String.format("DELETE FROM %s WHERE %s = '%s';", versionTableName, primaryKey, key.get());
+      stmt.execute(sqlDelete);
+    } catch (SQLException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+  @Override
+  public void onStartVersionIngestion(boolean isCurrentVersion) {
+    try (Connection connection = DriverManager.getConnection(duckDBUrl);
+        Statement stmt = connection.createStatement()) {
+      String createTableStatement =
+          AvroToSQL.createTableStatement(versionTableName, getOutputValueSchema(), primaryKeys, FAIL);
+      stmt.execute(createTableStatement);
+
+      if (isCurrentVersion) {
+        String createViewStatement =
+            String.format("CREATE OR REPLACE VIEW current_version AS SELECT * FROM %s;", versionTableName);
```

Comment:
similar nit, let's static constant these strings.

Reply:
ALSO, take a look at using prepared statements for all of these (this is to prevent sql injection vulnerabilities).

Reply:
I looked into this and prepared statements only work to inject values, IIUC, not to inject table names.